### DISCUSSION TITLE
Deprecation warning is misleading, should be changed

### DISCUSSION THREAD

**Discussion by OliverKunc** [clangd/clangd#2029] [Ideas]
Hi,

not sure whether this belongs here, please tell me if there is a better place for this.

When I call a constructor of MyClass and that constructor has a deprecation warning, then clangd prints:

> `'MyClass' is deprecated: <my deprecation warning concerning only a certain constructor>`

I find this misleading. The class itself is not deprecated.
My suggestion would be to replace it with

> `'MyClass' has deprecation warning: <my deprecation warning concerning only a certain constructor>`

or

> `Deprecation warning for 'MyClass': <my deprecation warning concerning only a certain constructor>`

or something similar.

**Comment by HighCommander4** [comment]
> please tell me if there is a better place for this.

Could you file it at https://github.com/llvm/llvm-project/issues/ please, with a code example?

The warning is produced by the clang frontend*, and clangd just surfaces it over LSP.

(* I think; it's hard to be sure without a code example. The other possibility is it's produced by clang-tidy, for which https://github.com/llvm/llvm-project/issues/ is also the right place to report.)
